evcache-zipkin-tracing/src/main/java/com/netflix/evcache/EVCacheTracingEventListener.java

```diff
@@ -0,0 +1,161 @@
+package com.netflix.evcache;
+
+import brave.Span;
+import brave.Tracer;
+import com.netflix.evcache.event.EVCacheEvent;
+import com.netflix.evcache.event.EVCacheEventListener;
+import com.netflix.evcache.pool.EVCacheClient;
+import com.netflix.evcache.pool.EVCacheClientPoolManager;
+import net.spy.memcached.CachedData;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.StringJoiner;
+
+/** Adds tracing tags for EvCache calls. */
+public class EVCacheTracingEventListener implements EVCacheEventListener {
+
+  public static String EVCACHE_SPAN_NAME = "evcache";
+
+  private static Logger logger = LoggerFactory.getLogger(EVCacheTracingEventListener.class);
+
+  private static String CLIENT_SPAN_ATTRIBUTE_KEY = "clientSpanAttributeKey";
+
+  private final Tracer tracer;
+
+  public EVCacheTracingEventListener(EVCacheClientPoolManager poolManager, Tracer tracer) {
+    poolManager.addEVCacheEventListener(this);
+    this.tracer = tracer;
+  }
+
+  @Override
+  public void onStart(EVCacheEvent e) {
+    try {
+      Span clientSpan =
+          this.tracer.nextSpan().kind(Span.Kind.CLIENT).name(EVCACHE_SPAN_NAME).start(e.getStartTime());
+
+      String appName = e.getAppName();
+      this.safeTag(clientSpan, EVCacheTracingTags.APP_NAME, appName);
+
+      String cacheNamePrefix = e.getCacheName();
+      this.safeTag(clientSpan, EVCacheTracingTags.CACHE_NAME_PREFIX, cacheNamePrefix);
+
+      String call = e.getCall().name();
+      this.safeTag(clientSpan, EVCacheTracingTags.CALL, call);
+
+      /**
+       * Note - e.getClients() returns a list of clients associated with the EVCacheEvent.
+       *
+       * <p>Read operation will have only 1 EVCacheClient as reading from just 1 instance of cache
+       * is sufficient. Write operations will have appropriate number of clients as each client will
+       * attempt to write to its cache instance.
+       */
+      String serverGroup;
+      StringJoiner serverGroups = new StringJoiner(",", "[", "]");
+      for (EVCacheClient client : e.getClients()) {
+        serverGroup = client.getServerGroupName();
+        if (StringUtils.isNotBlank(serverGroup)) {
+          serverGroups.add("\"" + serverGroup + "\"");
+        }
+      }
```

Comment:
This could apply to any tag value that will be of type list. May be this can be moved to the shared library. Also the `safeTag` method. More than code use, this will bring consistency in how the data is encoded on to the tags.